amundsen_application/static/js/components/ProfilePage/index.tsx

```diff
@@ -96,43 +106,68 @@ export class ProfilePage extends React.Component<ProfilePageProps, ProfilePageSt
   };
 
 
-  getTabContent = (resource: Resource[], source: string, label: string) => {
-    // TODO: consider moving logic for empty content into Tab component
-    if (resource.length === 0) {
-      return (
-        <div className="empty-tab-message body-placeholder">
-          User has no { label } resources.
-        </div>
-      );
-    }
+  generateTabContent = (resource: ResourceType) => {
+    const { bookmarks = [], own = [], read = [] } = this.props.resourceRelations[resource];
+    const resourceLabel = getDisplayNameByResource(resource);
     return (
-      <ResourceList
-        allItems={ resource }
-        source={ source }
-        itemsPerPage={ ITEMS_PER_PAGE }
-      />
+      <>
+        <ResourceList
+          allItems={ own }
+          itemsPerPage={ ITEMS_PER_PAGE }
+          paginate={ false }
+          source={ OWNED_SOURCE }
+          title={`${OWNED_TITLE_PREFIX} (${own.length})`}
+          customFooterText={`${FOOTER_TEXT_PREFIX} ${own.length} ${OWNED_LABEL} ${resourceLabel}`}
+          customEmptyText={`${EMPTY_TEXT_PREFIX} ${OWNED_LABEL} ${resourceLabel}.`}
```

Comment:
I'm kind of on the fence about this, but a helper function might help here.

Reply:
On the fence as well. I thought about helper functions for the strings but all it would do is concatenate a list of values with no extra logic so that seemed like a waste. I also thought of a helper function to render the components for each relationship but that would have come with so much boolean logic regarding what constants to use.

I will leave it as is unless I see a clear way to do something better.